lib/transaction.js

```diff
@@ -18,14 +20,18 @@ function mixinTransaction(PostgreSQL) {
    * @param cb
    */
   PostgreSQL.prototype.beginTransaction = function(isolationLevel, cb) {
+    var connector = this;
     debug('Begin a transaction with isolation level: %s', isolationLevel);
     this.pg.connect(function(err, connection, done) {
       if (err) return cb(err);
       connection.autorelease = done;
       connection.query('BEGIN TRANSACTION ISOLATION LEVEL ' + isolationLevel,
         function(err) {
           if (err) return cb(err);
-          cb(null, connection);
+          var tx = new Transaction(connector, connection);
+          tx.txId = uuid.v1();
+          connection.txId = tx.txId;
+          cb(null, tx);
```

Comment:
I don't think this is the case, as if you look at `loopback-conenctor` the commit method passes the connection of the transaction to the connector:

https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L48

And moreover, if you debug the code at line 76, you can see that the connection that is passed in to `releaseConnection` is definitely the connection that has `autorelease()` attached to it.

Calling back with transaction instead of the connection should make essentially no difference, as `loopback-connector` will immediately turn it into a `Transaction` if it is not already an instance of `Transaction`:

https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L88

I simply skip this step so we can attach the id to both the connection and the transaction.